tests/test_fits.py

```diff
@@ -68,3 +69,28 @@ async def draws(random_seed: Optional[int] = None) -> List[Union[int, float]]:
     # look at all draws
     assert np.allclose(draws1, draws2)
     assert not np.allclose(draws1, draws3)
+
+
+@pytest.mark.asyncio
+async def test_fits_vector_sizes(api_url: str) -> None:
+    """Simple test of sampling with zero and non-zero vector sizes."""
+
+    num_samples = num_warmup = 5000
+    payload = {
+        "function": "stan::services::sample::hmc_nuts_diag_e_adapt",
+        "num_samples": num_samples,
+        "num_warmup": num_warmup,
+    }
+    param_name = "z.1"
+    draws = await helpers.sample_then_extract(api_url, program_code_vector, payload, param_name)
+    assert len(draws) == num_samples, (len(draws), num_samples)
+    assert -0.01 < statistics.mean(draws) < 0.01
+
+    param_name = "z.2"
+    draws = await helpers.sample_then_extract(api_url, program_code_vector, payload, param_name)
+    assert len(draws) == num_samples, (len(draws), num_samples)
+    assert -0.01 < statistics.mean(draws) < 0.01
+
+    param_name = "x.1"
+    with pytest.raises(KeyError, match="No draws found for parameter `x.1`."):
+        await helpers.sample_then_extract(api_url, program_code_vector, payload, param_name)
```

Comment:
I'd suggest adding a random seed so the draws are cached (which will speed up the test). I'd also suggest lowering the num_samples to 500 so the test goes faster.

And I'd request that the `statistics.mean` tests be dropped or moved to new tests. Two reasons for this:
1. We test the characteristics of draws in other tests. This test should be about the zero-length vector parameter being present or absent in the draws, right?
2. These tests could, in theory, randomly fail if we get an unusual set of draws.

If you disagree, I'm happy to discuss this further. We can also find someone else to have a look at this.

Reply:
No disagreement here, I'm into keeping tests clean